packages/itwinui-css/src/bridge.scss

```diff
@@ -0,0 +1,513 @@
+// Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+// See LICENSE.md in the project root for license terms and full copyright notice.
+[data-iui-bridge='true'] {
+  * {
```

Comment:
We should specifically target the element where these variables are being set (i.e. `.iui-root`) instead of `*`.

Reply:
However, this does make me think: the attribute should be set on the same element that has all the variables.

So can't we target `.iui-root[data-iui-bridge="true"]`?

Reply:
> I realized that it was happening because the iTwinUI-react variables were actually unlayered and so taking priority over the bridge variable overrides.

This sounds like a bug that we should fix. All our CSS should be layered.

Reply:
After investigating this, it's not a bug in our package but rather in the usage. It's happening because we are importing `@itwin/itwinui-variables` in a JS file (outside a layer).

https://github.com/iTwin/iTwinUI/blob/330361346832306f81a63eff04d659adba47faf5/apps/react-workshop/.ladle/components.tsx#L13-L15

We can fix this by moving it into `global.css` and importing it into a layer.

Reply:
Try not to remove code without investigating why it was added. The variables import was necessary for the background switcher, because the `<div class="ladle-background">` element is rendered outside the ThemeProvider.

![](https://github.com/user-attachments/assets/39b5774f-50f6-4c46-886c-4d98281e8781)

Reply:
> To confirm, what do you mean by "CSS modules" here? Do we mean the `_iui3` prefix transformation?

Yes, that's what I mean. We use [Vite's CSS modules](https://vite.dev/guide/features.html#css-modules) feature for transforming the `styles.module.css` file.

We need to ensure that everything in the `@itwin/itwinui-react/styles.css` file is scoped properly. So the loose `[data-iui-bridge='true']` selector needs to changed to `.iui-root[data-iui-bridge='true']` (which will then get transformed to something like `._iui3160-root[data-iui-bridge='true']`.

Reply:
This is resolved in [`4ccd98c`](https://github.com/iTwin/iTwinUI/pull/2373/commits/4ccd98cd3cd0658faeead1b6d18272dcd92a4a7a)